event_database/cloud_functions/notional-transferred-from.go

```diff
@@ -0,0 +1,164 @@
+			// "0 daysAgo end" is 23:59:59 of the current day (the future)
+
+			// calculate the start and end times for the query
+			hoursAgo := (24 * daysAgo)
+			daysAgoDuration := -time.Duration(hoursAgo) * time.Hour
+			n := now.Add(daysAgoDuration)
+			year := n.Year()
+			month := n.Month()
+			day := n.Day()
+			loc := n.Location()
+
+			start := time.Date(year, month, day, 0, 0, 0, 0, loc)
+			end := time.Date(year, month, day, 23, 59, 59, maxNano, loc)
+
+			dateStr := start.Format("2006-01-02")
+
+			muTransfersFromCache.Lock()
+			// check to see if there is cache data for this date/query
+			if _, ok := transfersFromCache.Daily[dateStr]; ok && useCache(dateStr) {
+				// have a cache for this date
+				if daysAgo >= 1 {
+					// only use the cache for yesterday and older
+					muTransfersFromCache.Unlock()
+					return
+				}
+			} else {
+				// no cache for this query, initialize the map
+				transfersFromCache.Daily[dateStr] = map[string]float64{"*": 0}
+			}
+			muTransfersFromCache.Unlock()
+
+			queryResult := fetchTransferRowsInInterval(tbl, ctx, prefix, start, end)
+
+			// iterate through the rows and increment the amounts
+			for _, row := range queryResult {
+				if _, ok := tokensToSkip[row.TokenAddress]; ok {
+					// skip blacklisted token
+					continue
+				}
+				if _, ok := transfersFromCache.Daily[dateStr][row.LeavingChain]; !ok {
+					transfersFromCache.Daily[dateStr][row.LeavingChain] = 0
+				}
+				transfersFromCache.Daily[dateStr]["*"] = transfersFromCache.Daily[dateStr]["*"] + row.Notional
+				transfersFromCache.Daily[dateStr][row.LeavingChain] = transfersFromCache.Daily[dateStr][row.LeavingChain] + row.Notional
+			}
+		}(tbl, ctx, prefix, daysAgo)
+	}
+	intervalsWG.Wait()
+
+	// having consistent keys in each object is helpful for clients, explorer GUI
+	transfersFromCache.Total = 0
+	seenChainSet := map[string]bool{}
+	for _, chains := range transfersFromCache.Daily {
+		for chain, amount := range chains {
+			seenChainSet[chain] = true
+			if chain == "*" {
+				transfersFromCache.Total += amount
+			}
+		}
+	}
+	for date, chains := range transfersFromCache.Daily {
+		for chain := range seenChainSet {
+			if _, ok := chains[chain]; !ok {
+				transfersFromCache.Daily[date][chain] = 0
+			}
+		}
+	}
+
+	persistInterfaceToJson(ctx, transfersFromFilePath, &muTransfersFromCache, transfersFromCache)
+}
+
+// finds the value that has been transferred from each chain
+func ComputeNotionalTransferredFrom(w http.ResponseWriter, r *http.Request) {
+	// Set CORS headers for the preflight request
+	if r.Method == http.MethodOptions {
+		w.Header().Set("Access-Control-Allow-Origin", "*")
+		w.Header().Set("Access-Control-Allow-Methods", "POST")
+		w.Header().Set("Access-Control-Allow-Headers", "Content-Type")
+		w.Header().Set("Access-Control-Max-Age", "3600")
+		w.WriteHeader(http.StatusNoContent)
+		return
+	}
+	// Set CORS headers for the main request.
+	w.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
If you moved setting `Access-Control-Allow-Origin` above the conditional for the OPTIONS request, you could remove that line from the conditional as wel.